### PR TITLE
filterbuyers and filterprops by characteristics

### PR DESCRIPTION
**Pull request by zsiggg**
Able to filter by multiple characteristics at one time (separate characteristics with semicolon)

A characteristic is considered a match if a person's or property's characteristic **contains** it (instead of an exact match), after trimming whitespace and ignoring the case.

Addresses #99 and #105

### CODE REVIEW

File: src/main/java/seedu/address/model/characteristics/Characteristics.java [hunk 1]
```diff
@@ -45,12 +45,21 @@ public static boolean isValidCharacteristics(String test) {
...
-                .anyMatch(c -> c.contains(characteristic));
+                .anyMatch(c -> c.equalsIgnoreCase(characteristic));
+    }
+
+    /**
+     * Returns true if a given {@code Characteristic} contains a characteristic that
+     * is also found in this {@code Characteristic}
+     */
+    public boolean containsCharacteristics(Characteristics other) {
+        return Arrays.stream(other.characteristicsArr)
```
Comment (chngchngchng) [inline src/main/java/seedu/address/model/characteristics/Characteristics.java:61]:
```suggestion
        return Arrays.stream(other.getCharacteristicsArr())
```
Can consider adding a getter for the characteristics array!

File: src/main/java/seedu/address/model/characteristics/Characteristics.java [hunk 2]
```diff
@@ -45,12 +45,21 @@ public static boolean isValidCharacteristics(String test) {
...
         return Arrays.stream(characteristicsArr)
-                .anyMatch(c -> c.contains(characteristic));
+                .anyMatch(c -> c.equalsIgnoreCase(characteristic));
+    }
+
+    /**
+     * Returns true if a given {@code Characteristic} contains a characteristic that
+     * is also found in this {@code Characteristic}
+     */
+    public boolean containsCharacteristics(Characteristics other) {
```
Comment (chngchngchng) [inline src/main/java/seedu/address/model/characteristics/Characteristics.java:60]:
```suggestion
    public boolean containsAnyGivenCharacteristics(Characteristics other) {
```
Perhaps renaming the function could differentiate it more from `containsCharacteristic`.

File: src/main/java/seedu/address/model/property/FilterPropsByCharacteristicsPredicate.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.model.characteristics.Characteristics;
+
+/**
+ * Tests that a given {@code Property}'s {@code DesiredCharacteristics} contains the given characteristic.
+ */
+public class FilterPropsByCharacteristicsPredicate extends AbstractFilterPropsPredicate {
+
+    private static final String CHARACTERISTIC_DELIMITER = ";";
```
Comment (chngchngchng) [inline src/main/java/seedu/address/model/property/FilterPropsByCharacteristicsPredicate.java:12]:
```suggestion

```
Redundant line of code!

**Review by chngchngchng — APPROVED** [review]
LGTM! Just some minor observations and potential changes.

File: src/main/java/seedu/address/model/property/FilterPropsByCharacteristicsPredicate.java (hunk 3 above)
Comment (zsiggg) [reply src/main/java/seedu/address/model/property/FilterPropsByCharacteristicsPredicate.java]: Removed in FilterBuyersByCharacteristicsPredicate as well

**Pull request merged**